packages/core-react/src/hooks/useDisconnect.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useContext } from 'react';
+import { useAccount, useDisconnect as useDisconnectWagmi } from 'wagmi';
+import { useConnectorInfo } from './useConnectorInfo';
+import { AcceptTermsModalContext } from '../context/acceptTermsModal';
+
+export const useForceDisconnect = () => {
+  const { disconnect } = useDisconnectWagmi();
+  const { acceptTermsModal } = useContext(AcceptTermsModalContext);
+
+  const forceDisconnect = useCallback(() => {
+    disconnect();
+    acceptTermsModal.setVisible(false);
+  }, [acceptTermsModal, disconnect]);
```

Comment:
This callback will change every time `acceptTermsModal` changes state. Extract `setVisible` and pass it directly, react state setters don't change identity.

Reply:
Reworked